src/Libraries/CoreNodeModels/CustomSelectionItem.cs

```diff
@@ -0,0 +1,165 @@
+using System;
+using System.ComponentModel;
+using System.Windows.Input;
+using Newtonsoft.Json;
+
+namespace CoreNodeModels
+{
+    /// <summary>
+    /// Represents an Enum item
+    /// </summary>
+    public class CustomSelectionItem
+    {
+        public string Name { get; set; }
+
+        public string Value { get; set; }
+
+        public CustomSelectionItem()
+        {
+
+        }
+    }
+
+
+    public class CustomSelectionItemViewModel : INotifyPropertyChanged, IDataErrorInfo
+    {
+        private readonly CustomSelectionItem customSelectionItem;
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        #region Properties
+
+        [JsonIgnore]
+        public Func<CustomSelectionItem, bool> IsUnique { get; set; }
+
+        [JsonIgnore]
+        public Action ItemChanged { get; set; }
+
+        [JsonIgnore]
+        public Action<CustomSelectionItemViewModel> RemoveRequested { get; set; }
+
+        public string Name
+        {
+            get
+            {
+                return customSelectionItem.Name;
+            }
+            set
+            {
+                customSelectionItem.Name = value;
+
+                OnPropertyChanged("Name");
+                OnPropertyChanged("IsValid");
+
+                if (ItemChanged != null)
+                    ItemChanged();
+            }
+        }
+
+        public string Value
+        {
+            get
+            {
+                return customSelectionItem.Value;
+            }
+            set
+            {
+                customSelectionItem.Value = value;
+                OnPropertyChanged("Value");
+
+                if (ItemChanged != null)
+                    ItemChanged();
+            }
+        }
+
+        [JsonIgnore]
+        public bool IsValid
+        {
+            get
+            {
+                if (string.IsNullOrWhiteSpace(Name))
+                    return false;
+
+                if (IsUnique != null && !IsUnique(customSelectionItem))
+                    return false;
+
+                return true;
+            }
+        }
+
+        #endregion
+
+        #region IDataErrorInfo
+
+        [JsonIgnore]
+        public string Error
+        {
+            get
+            {
+                return string.Empty;
+            }
+        }
+
+        [JsonIgnore]
+        public string this[string columnName]
+        {
+            get
+            {
+                if (columnName.Equals("Name"))
+                {
+                    if (string.IsNullOrWhiteSpace(Name))
+                        return "Name cannot be empty";
```

Comment:
this text needs to be stored in a resx so it can be localized. See examples in other nodes like:
https://github.com/DynamoDS/Dynamo/blob/102bd97db0ccc33ada9ff346178ae6da6de34607/src/Libraries/UnitsUI/UnitsUI.cs#L128

Reply:
This assessor and other things related to IDataErrorInfo are not really necessary and will be removed